### ISSUE TITLE
Capturing the success or failure of DeleteListener and ApplyTransformListener

### ISSUE THREAD

**Issue by srinathgit** [marklogic/data-movement#96]
The code below tries to delete a non existent document in db. 
1. I am not sure of the HTTP code returned in this case but is there any way to capture its success or failure  (whether a transform has been applied or not and in this case if the document was deleted or not) ? 

2.Additionally,  Is it correct that the onQueryFailure() method is never called when QHB is created using an iterator of uris ? 

```
@Test
    public void deleteNonExistentDoc() throws Exception{

        HashSet<String> urisList = new HashSet<>();
        //add a non existent doc uri
        urisList.add("/abc/nonexistent");

        AtomicInteger successDocs = new AtomicInteger();
        HashSet<String> uris2 = new HashSet<>();
        StringBuffer failures2 = new StringBuffer();


        QueryHostBatcher deleteBatcher = dmManager.newQueryHostBatcher(urisList.iterator())
        .withBatchSize(11)
        .withThreadCount(11)
        .onUrisReady(new DeleteListener())
        .onUrisReady((client, batch) -> successDocs.addAndGet(batch.getItems().length))
        .onUrisReady((client, batch) -> uris2.addAll(Arrays.asList(batch.getItems())))
        .onQueryFailure( (client, throwable) -> {
            throwable.printStackTrace();
            failures2.append("ERROR:[" + throwable + "]\n");
          });

        JobTicket delTicket = dmManager.startJob(deleteBatcher);
        deleteBatcher.awaitTermination(Long.MAX_VALUE, TimeUnit.DAYS);
        dmManager.stopJob(delTicket);

    //      if ( failures2.length() > 0 ) fail(failures2.toString());
        Assert.assertTrue(dbClient.newServerEval().xquery(query1).eval().next().getNumber().intValue() == 0);
    }
```

**Comment by sammefford** [comment]
In this case the HTTP code returned is a 204, so no exception is thrown.  You can see this in the wire trace by adding this line to the beginning of your code:

    System.setProperty("org.apache.commons.logging.simplelog.log.org.apache.http.wire", "debug");

If you'd like to trigger an exception, try this instead:

    .onUrisReady((client, batch) -> client.newDocumentManager().read(batch.getItems()[0], new BytesHandle()))

You can see with that that onQueryFailure is called.  However, I'm not sure the callback name "onQueryFailure" is appropriate here, so I logged a task to think through that: https://github.com/marklogic/data-movement/issues/97

**Comment by srinathgit** [comment]
Shipping this bug as the issue regarding callback name is addressed in #97
